### PR TITLE
sortLevel feature added into configState

### LINKED ISSUE #269: Option to have cards in a draw sorted by difficulty
Cards are currently presented shuffled in totally random order. It would be nice to have an option for them to instead be sorted by difficulty level. (And maybe sorted by other comparisons in the future?)

### PR DESCRIPTION
**Pull request by paulitup**
Added a checkbox within General Settings to sort charts by difficulty level.

Fixes #269

### CODE REVIEW

File: src/drawn-set.tsx [hunk 1]
```diff
@@ -24,6 +24,10 @@ interface Props {
 
 function ChartList() {
   const charts = useDrawing((d) => d.charts);
+  const sortByLevel = useConfigState((s) => s.sortByLevel);
+  if (sortByLevel) {
+    charts.sort((a, b) => a.level - b.level);
+  }
```
Comment (noahm) [inline src/drawn-set.tsx:30]: I don't want this particular sorting to happen in the component responsible for displaying the drawings. The order of cards in a drawing may start out sorted, but may also change as players take action on them.

Instead, look in `card-draw.ts` near the end. You'll find that after all the logic of picking the cards it shuffles them as a final step before returning. This new feature should be sorting by difficulty there, at the moment a new set is drawn.

**Review by noahm — CHANGES_REQUESTED** [review]
Great job! Everything here is exactly as I would do it, save for one specific detail. In this case sorting should not be a matter of display logic, but of the draw procedure itself.

File: src/drawn-set.tsx (hunk 1 above)
Comment (paulitup) [reply src/drawn-set.tsx]: Should be updated! Thanks for the review! @noahm

File: src/card-draw.ts [hunk 2]
```diff
@@ -218,7 +218,7 @@ export function draw(gameData: GameData, configData: ConfigState): Drawing {
 
   return {
     id: `draw-${nanoid(10)}`,
-    charts: shuffle(drawnCharts),
+    charts: configData.sortByLevel ? drawnCharts.sort((a, b) => a.level - b.level) : shuffle(drawnCharts),
```
Comment (noahm) [inline src/card-draw.ts:221]: This doesn't match prettier formatting, and fails the vercel deploy

**Review by noahm — CHANGES_REQUESTED** [review]
One last nit to fix and this is ready to merge!

**Pull request merged**